### ISSUE TITLE
AttributeError: 'MARBLComponent' object has no attribute 'time_step'

### ISSUE THREAD

**Issue by TomNicholas** [CWorthy-ocean/C-Star#14]
When running the notebook I got this error (I ran some cells out of order)

```python
marbl_component=cstar.MARBLComponent(
                    base_model=marbl_base_model,
                    additional_code=marbl_additional_code
)
marbl_component   
```
```python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
File [~/miniconda3/envs/cstar_environment/lib/python3.12/site-packages/IPython/core/formatters.py:711](http://localhost:8888/lab/tree/~/miniconda3/envs/cstar_environment/lib/python3.12/site-packages/IPython/core/formatters.py#line=710), in PlainTextFormatter.__call__(self, obj)
    704 stream = StringIO()
    705 printer = pretty.RepresentationPrinter(stream, self.verbose,
    706     self.max_width, self.newline,
    707     max_seq_length=self.max_seq_length,
    708     singleton_pprinters=self.singleton_printers,
    709     type_pprinters=self.type_printers,
    710     deferred_pprinters=self.deferred_printers)
--> 711 printer.pretty(obj)
    712 printer.flush()
    713 return stream.getvalue()

File [~/miniconda3/envs/cstar_environment/lib/python3.12/site-packages/IPython/lib/pretty.py:419](http://localhost:8888/lab/tree/~/miniconda3/envs/cstar_environment/lib/python3.12/site-packages/IPython/lib/pretty.py#line=418), in RepresentationPrinter.pretty(self, obj)
    408                         return meth(obj, self, cycle)
    409                 if (
    410                     cls is not object
    411                     # check if cls defines __repr__
   (...)
    417                     and callable(_safe_getattr(cls, "__repr__", None))
    418                 ):
--> 419                     return _repr_pprint(obj, self, cycle)
    421     return _default_pprint(obj, self, cycle)
    422 finally:

File [~/miniconda3/envs/cstar_environment/lib/python3.12/site-packages/IPython/lib/pretty.py:787](http://localhost:8888/lab/tree/~/miniconda3/envs/cstar_environment/lib/python3.12/site-packages/IPython/lib/pretty.py#line=786), in _repr_pprint(obj, p, cycle)
    785 """A pprint that just redirects to the normal repr function."""
    786 # Find newlines and replace them with p.break_()
--> 787 output = repr(obj)
    788 lines = output.splitlines()
    789 with p.group():

File [~/Documents/Work/Code/C-Star/cstar_ocean/component.py:148](http://localhost:8888/lab/tree/~/Documents/Work/Code/C-Star/cstar_ocean/component.py#line=147), in Component.__repr__(self)
    147 def __repr__(self):
--> 148     return self.__str__()

File [~/Documents/Work/Code/C-Star/cstar_ocean/component.py:104](http://localhost:8888/lab/tree/~/Documents/Work/Code/C-Star/cstar_ocean/component.py#line=103), in Component.__str__(self)
    101 base_str += "\n" + "-" * (len(name) + 7)
    103 # Attrs
--> 104 base_str += f"\ntime_step: {self.time_step} seconds"
    105 base_str += "\nBuilt from: "
    107 NAC = 0 if self.additional_code is None else 1

AttributeError: 'MARBLComponent' object has no attribute 'time_step'
marbl_component
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
File [~/miniconda3/envs/cstar_environment/lib/python3.12/site-packages/IPython/core/formatters.py:711](http://localhost:8888/lab/tree/~/miniconda3/envs/cstar_environment/lib/python3.12/site-packages/IPython/core/formatters.py#line=710), in PlainTextFormatter.__call__(self, obj)
    704 stream = StringIO()
    705 printer = pretty.RepresentationPrinter(stream, self.verbose,
    706     self.max_width, self.newline,
    707     max_seq_length=self.max_seq_length,
    708     singleton_pprinters=self.singleton_printers,
    709     type_pprinters=self.type_printers,
    710     deferred_pprinters=self.deferred_printers)
--> 711 printer.pretty(obj)
    712 printer.flush()
    713 return stream.getvalue()

File [~/miniconda3/envs/cstar_environment/lib/python3.12/site-packages/IPython/lib/pretty.py:419](http://localhost:8888/lab/tree/~/miniconda3/envs/cstar_environment/lib/python3.12/site-packages/IPython/lib/pretty.py#line=418), in RepresentationPrinter.pretty(self, obj)
    408                         return meth(obj, self, cycle)
    409                 if (
    410                     cls is not object
    411                     # check if cls defines __repr__
   (...)
    417                     and callable(_safe_getattr(cls, "__repr__", None))
    418                 ):
--> 419                     return _repr_pprint(obj, self, cycle)
    421     return _default_pprint(obj, self, cycle)
    422 finally:

File [~/miniconda3/envs/cstar_environment/lib/python3.12/site-packages/IPython/lib/pretty.py:787](http://localhost:8888/lab/tree/~/miniconda3/envs/cstar_environment/lib/python3.12/site-packages/IPython/lib/pretty.py#line=786), in _repr_pprint(obj, p, cycle)
    785 """A pprint that just redirects to the normal repr function."""
    786 # Find newlines and replace them with p.break_()
--> 787 output = repr(obj)
    788 lines = output.splitlines()
    789 with p.group():

File [~/Documents/Work/Code/C-Star/cstar_ocean/component.py:148](http://localhost:8888/lab/tree/~/Documents/Work/Code/C-Star/cstar_ocean/component.py#line=147), in Component.__repr__(self)
    147 def __repr__(self):
--> 148     return self.__str__()

File [~/Documents/Work/Code/C-Star/cstar_ocean/component.py:104](http://localhost:8888/lab/tree/~/Documents/Work/Code/C-Star/cstar_ocean/component.py#line=103), in Component.__str__(self)
    101 base_str += "\n" + "-" * (len(name) + 7)
    103 # Attrs
--> 104 base_str += f"\ntime_step: {self.time_step} seconds"
    105 base_str += "\nBuilt from: "
    107 NAC = 0 if self.additional_code is None else 1

AttributeError: 'MARBLComponent' object has no attribute 'time_step'
```
This shouldn't be possible, and IMO is another argument for splitting out the discretization as a separate class (#15)

**Comment by TomNicholas** [comment]
It also gives the same error if I restart and run the notebook top-to-bottom.

**Comment by dafyddstephenson** [comment]
Addressed in PR #16

**Comment by TomNicholas** [comment]
Closed by #16
